### PR TITLE
Allow ./ prefix for file

### LINKED ISSUE #96243: Explorer: new file does not allow ./ syntax?
image

I think it is fair to prevent creation of files outside the workspace root, but `./` seems pretty safe to me.

### PR DESCRIPTION
**Pull request by TroublingStone**
I changed Invalid file chars regexes for both UNIX and Windows to allow `./`, but not `/`.
This PR fixes #96243

### CODE REVIEW

**Comment by isidorn** [comment]
Thank you for this PR, however we are currently in endgame week which means we do not take new features and only test. Thus assigning this to May so I can review it next week. Thanks

File: src/vs/base/common/extpath.ts [hunk 1]
```diff
@@ -139,8 +139,8 @@ export function isUNC(path: string): boolean {
 }
 
 // Reference: https://en.wikipedia.org/wiki/Filename
-const WINDOWS_INVALID_FILE_CHARS = /[\\/:\*\?"<>\|]/g;
-const UNIX_INVALID_FILE_CHARS = /[\\/]/g;
+const WINDOWS_INVALID_FILE_CHARS = /\.\.\/|\\|(?<!\.)\/|:|\*|\?|"|<|>|\|/g;
```
Comment (isidorn) [inline src/vs/base/common/extpath.ts:142]: This regexes grew a lot. Is it possible to have this in simpler form?
Or to fix it some other way?

**Review by isidorn — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
@LilJohny thanks again for this PR. Event though it fixes the issue it overcomplicates some regexes imho. To only cover this corner case I would not accept this PR which doubles the regex in complexity.

File: src/vs/base/common/extpath.ts (hunk 1 above)
Comment (LilJohny) [reply src/vs/base/common/extpath.ts:142]: Do you mean Windows regex only or both of them?

Comment (isidorn) [reply src/vs/base/common/extpath.ts:142]: Both trippled in size.

Comment (LilJohny) [reply src/vs/base/common/extpath.ts]: @isidorn I`ve managed to shorten Windows regex, but there is no way to reduce Unix regex. Do you think that separate if statement will do better?

**Pull request closed without merging**